### DISCUSSION TITLE
Comparison of Kafka GUI Tools

### DISCUSSION THREAD

**Discussion by xarx00** [tchiotludo/akhq#805] [Ideas]
FYI, a fresh [article comparing 10 Kafka GUI Tools](https://towardsdatascience.com/overview-of-ui-tools-for-monitoring-and-management-of-apache-kafka-clusters-8c383f897e80), including AKHQ. There is a nice overview table in the article. In https://github.com/tchiotludo/akhq/discussions/632 you were asking what features are missing in AKHQ. Of course, the article is opinionated, not that I agree with everything written in there...

Before, I was using Conduktor. It is a very good application (far from perfect), but AKHQ has in many respects similar look an feel, IMO. Where is AKHQ lacking is the responsiveness (Conduktor caches many data, of course, its a desktop application), message production (naturally, it's a new feature), and UI navigation. Concerning UI navigation, it would help

- if direct navigation from a topic to its subject/schema (and vice versa) would be possible - When I am on a topic detail, I have to click on Schema registry, enter the search string and find the subject, click on detail. And to navigate back, I have to do the same with Topics.
- If the message production dialog stays open after producing a message - When I need to produce multiple messages, each time I have to open the production dialog, fill all fields, and copy the message from clipboard.
- if filters kept their values - For instance, whenever I click on Topics, I have to enter the Search string again and again.
-  if lists kept their content - For instance, when navigating back, the filters get reset and everything is downloaded from the server again. This doesn't help the responsiveness.

But in general, I like AKHQ, I think it is a good application, and it still evolves :-).

**Comment by tchiotludo** [comment]
Thanks for sharing this one :) 
I open the matrix and I see that AKHQ is the only one with Bad UI and I want to close the page :smile: (especially comparing some other on the matrix. 

You are true about responsiveness, I'm aware of that but I never found time to increase the performance except on the topic page. 


> if direct navigation from a topic to its subject/schema (and vice versa) would be possible - When I am on a topic detail, I have to click on Schema registry, enter the search string and find the subject, click on detail. And to navigate back, I have to do the same with Topics.

Great Idea, open an issue about that but would be a kind of complicated without caching the registry schema. 

> If the message production dialog stays open after producing a message - When I need to produce multiple messages, each time I have to open the production dialog, fill all fields, and copy the message from clipboard.

Same here, and this one is simple, maybe a PR that remove the redirect ?  

> if filters kept their values - For instance, whenever I click on Topics, I have to enter the Search string again and again.
> if lists kept their content - For instance, when navigating back, the filters get reset and everything is downloaded from the server again. This doesn't help the responsiveness.

I also notice that, it can be easy, just be sure all the param are on query string and kept the last url for every list on session storage, It will fix both of them.

For the ui, my main concern is that the ui was full rewrite in React and I'm not fluent with React, so it's take more time for me to tune the UI. I really hope 1 day to have some contributors that can help on the ui. 

Thanks for your feedback, really appreciate

**Comment by xarx00** [comment]
Actually, I think that the UI is pretty good - with respect to that I don't like web apps much in general, and I prefer classical desktop UI to the Metro-like UIs or mobile UIs on Windows. And in this respect the article is surely opinionated, perhaps the author prefers classical clickable web apps, not web apps with active UI.

> You are true about responsiveness, I'm aware of that but I never found time to increase the performance except on the topic page.

In my opinion, the application itself seems pretty swift. What slows it down are the steady data readings from the server.

> For the ui, my main concern is that the ui was full rewrite in React and I'm not fluent with React, so it's take more time for me to tune the UI.

I'm not a web developer, so React (and Node) is all new to me.
